spec/array.spec.ts

```diff
@@ -2,8 +2,8 @@ import { types } from "../src";
 
 test("can create an array of simple types", () => {
   const arrayType = types.array(types.string);
-  expect(arrayType.createReadOnly()).toEqual([]);
-  expect(arrayType.createReadOnly(["a", "b"])).toEqual(["a", "b"]);
+  expect(arrayType.createReadOnly().toJSON()).toEqual([]);
```

Comment:
Are these no longer necessary?

Reply:
Nah, they're needed unfortunately :(

```
    expect(received).toEqual(expected) // deep equality

    Compared values serialize to the same structure.
    Printing internal object structure without calling `toJSON` instead.

    - Expected  - 1
    + Received  + 1

    - Array []
    + QuickArray []
```

Different errors, but it's what we did in `map.spec.ts` too. Jest's `toEqual` matcher is unfortunately too picky in this instance :(

Fortunately I only had to do something like this in one place in the gadget repo. I was worried it would have been way more.